### PR TITLE
Add profile for udiskie

### PR DESCRIPTION
**Pull request by jose1711**

### CODE REVIEW

File: etc/udiskie.profile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+machine-id
+net none
+nogroups
+nonewprivs
+noroot
+notv
+nou2f
+novideo
+seccomp
+shell none
```
Comment (rusty-snake) [inline etc/udiskie.profile]: What about `protocol unix` ?

File: etc/udiskie.profile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc include disable-programs.inc
+include disable-xdg.inc
+
+caps.drop all
+machine-id
+net none
+nogroups
```
Comment (rusty-snake) [inline etc/udiskie.profile]: What about ~`nodbus` and~ `no3d`?

File: etc/udiskie.profile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+include globals.local
+
+# Allow python (blacklisted by disable-interpreters.inc)
+include allow-python3.inc
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc include disable-programs.inc
```
Comment (rusty-snake) [inline etc/udiskie.profile]: Only once ;)

File: etc/udiskie.profile [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Firejail profile for udiskie
+# Description: Removable disk automounter using udisks
+# This file is overwritten after every install/update
+# quiet
```
Comment (rusty-snake) [inline etc/udiskie.profile]: Remove.

File: etc/udiskie.profile [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+# Allow python (blacklisted by disable-interpreters.inc)
+include allow-python3.inc
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc include disable-programs.inc
+include disable-xdg.inc
+
```
Comment (rusty-snake) [inline etc/udiskie.profile]: What about `include whitelist-var-common.inc`?

and `whitelist`ing in general?

File: etc/udiskie.profile [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+caps.drop all
+machine-id
+net none
+nogroups
+nonewprivs
+noroot
+notv
```
Comment (rusty-snake) [inline etc/udiskie.profile]: What about `nosound`?

File: etc/udiskie.profile [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+nou2f
+novideo
+seccomp
+shell none
+tracelog
+
+private-cache
+private-dev
+private-tmp
```
Comment (rusty-snake) [inline etc/udiskie.profile]: Can we add `private` ?

File: etc/udiskie.profile [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+nonewprivs
+noroot
+notv
+nou2f
+novideo
+seccomp
+shell none
+tracelog
+
+private-cache
```
Comment (rusty-snake) [inline etc/udiskie.profile]: What about `private-bin` and `private-etc`?

File: etc/udiskie.profile (hunk 1 above)
Comment (jose1711) [reply etc/udiskie.profile]: added

File: etc/udiskie.profile (hunk 2 above)
Comment (jose1711) [reply etc/udiskie.profile]: `no3d` added

File: etc/udiskie.profile (hunk 7 above)
Comment (jose1711) [reply etc/udiskie.profile]: not sure. I tried but it requires quite a lot of utils to open associated file browser (`sed`, `cut`..)

Comment (rusty-snake) [reply etc/udiskie.profile]: You mean `private-bin`? Then leave it or add that what you found, but comment the line.

File: etc/udiskie.profile (hunk 8 above)
Comment (jose1711) [reply etc/udiskie.profile]: > What about `private-bin` and `private-etc`?

this apparently does not hurt: ```private-etc ld.so.cache,ld.so.preload,ld.so.conf,ld.so.conf.d,locale,locale.alias,locale.conf,localtime,alternatives,mime.types,xdg```

**Comment by rusty-snake** [comment]
Ohh no now there are two files `udiskie.profile` and `etc/udiskie.profile`

**Comment by rusty-snake** [comment]
Ohh no now there are two files `udiskie.profile` and `etc/udiskie.profile`

File: etc/udiskie.profile (hunk 8 above)
Comment (rusty-snake) [reply etc/udiskie.profile]: No `fonts`?

File: etc/udiskie.profile (hunk 2 above)
Comment (rusty-snake) [reply etc/udiskie.profile]: Move over `nogroups`.

**Comment by rusty-snake** [comment]
Great, only a few things left:
 * Update: `private-etc alternatives,ld.so.cache,ld.so.conf,ld.so.conf.d,ld.so.preload,locale,locale.alias,locale.conf,localtime,mime.types,xdg`
 * Update: `private-bin awk,cut,dbus-send,egrep,file,grep,head,python*,readlink,sed,sh,udiskie,uname,which,xdg-mime,xdg-open,xprop`
 * Remove: `udiskie.profile` (not `etc/udiskie.profile`)
 * Remove: Line 45 (blank line over `private-tmp`)
 * Move: `no3d` between `net none` and `nogroups`
 * Move: line 41-44 (`private-bin`) over `private-cache`

<details>
<summary>diffs for `etc/udiskie.profile`, `udiskie.profile` still needs to be removed</summary>

```diff
*** udiskie.profile.orig	2019-06-17 10:39:55.241861864 +0200
--- udiskie.profile.fixed	2019-06-17 10:39:17.099303997 +0200
*************** include whitelist-var-common.inc
*** 22,27 ****
--- 22,28 ----
  caps.drop all
  machine-id
  net none
+ no3d
  nogroups
  nonewprivs
  noroot
*************** nosound
*** 29,47 ****
  notv
  nou2f
  novideo
- no3d
  protocol unix
  seccomp
  shell none
  tracelog
  
! private-cache
! private-dev
! private-etc ld.so.cache,ld.so.preload,ld.so.conf,ld.so.conf.d,locale,locale.alias,locale.conf,localtime,alternatives,mime.types,xdg
! private-bin awk,cut,dbus-send,egrep,file,grep,head,python,python3,readlink,sed,sh,udiskie,uname,which,xdg-mime,xdg-open,xprop
! # add your configured file browser in udiskie.local, e. g.
  # private-bin nautilus
  # private-bin thunar
! 
  private-tmp
- 
--- 30,45 ----
  notv
  nou2f
  novideo
  protocol unix
  seccomp
  shell none
  tracelog
  
! private-bin awk,cut,dbus-send,egrep,file,grep,head,python*,readlink,sed,sh,udiskie,uname,which,xdg-mime,xdg-open,xprop
! # add your configured file browser in udiskie.local, e.g.
  # private-bin nautilus
  # private-bin thunar
! private-cache
! private-dev
! private-etc alternatives,ld.so.cache,ld.so.conf,ld.so.conf.d,ld.so.preload,locale,locale.alias,locale.conf,localtime,mime.types,xdg
  private-tmp
```

```diff
24a25
> no3d
32d32
< no3d
38,42c38,39
< private-cache
< private-dev
< private-etc ld.so.cache,ld.so.preload,ld.so.conf,ld.so.conf.d,locale,locale.alias,locale.conf,localtime,alternatives,mime.types,xdg
< private-bin awk,cut,dbus-send,egrep,file,grep,head,python,python3,readlink,sed,sh,udiskie,uname,which,xdg-mime,xdg-open,xprop
< # add your configured file browser in udiskie.local, e. g.
---
> private-bin awk,cut,dbus-send,egrep,file,grep,head,python*,readlink,sed,sh,udiskie,uname,which,xdg-mime,xdg-open,xprop
> # add your configured file browser in udiskie.local, e.g.
45c42,44
< 
---
> private-cache
> private-dev
> private-etc alternatives,ld.so.cache,ld.so.conf,ld.so.conf.d,ld.so.preload,locale,locale.alias,locale.conf,localtime,mime.types,xdg
47d45
< 
```

</details>

***
@jose1711 can you leave a comment when you do a force-push, because otherwise I don't get a notification.

**Comment by jose1711** [comment]
> diffs for `etc/udiskie.profile`, `udiskie.profile` still needs to be removed

Please where do you see these? I checked the PR and I can only see two files being changed (profile and `firecfg` config)

**Comment by rusty-snake** [comment]
> Please where do you see these?

No it is removed, but it was there:
* https://github.com/netblue30/firejail/commit/90ac2fad8346344306adc9c9573a573f693ad503#diff-9f3cb83906f34c36f28a4815b351fecb
 * https://github.com/netblue30/firejail/commit/90ac2fad8346344306adc9c9573a573f693ad503#diff-9f3cb83906f34c36f28a4815b351fecb
* https://github.com/netblue30/firejail/commit/0657e6ef09272915f1b4ce818de2a57b7a6f11d6#diff-9f3cb83906f34c36f28a4815b351fecb
 * https://github.com/netblue30/firejail/commit/0657e6ef09272915f1b4ce818de2a57b7a6f11d6#diff-9f3cb83906f34c36f28a4815b351fecb

**Comment by jose1711** [comment]
> No it is removed, but it was there:
> 
b83906f34c36f28a4815b351fecb)

OK, probably vanished somewhere in the middle of updates

**Comment by rusty-snake** [comment]
Merged, Thanks.

**Pull request merged**